src/vcpkg/visualstudio.cpp

```diff
@@ -146,15 +145,19 @@ namespace vcpkg::VisualStudio
             {
                 // We want lexically_normal(), but it is not available
                 // Correct root path might be 2 or 3 levels up, depending on if the path has trailing backslash.
-                auto common7_tools = vcpkg::u8path(*path_as_string);
+                Path common7_tools = *path_as_string;
                 if (common7_tools.filename().empty())
-                    maybe_append_path(common7_tools.parent_path().parent_path().parent_path(), version, check_cl);
-                else
-                    maybe_append_path(common7_tools.parent_path().parent_path(), version, check_cl);
+                {
+                    common7_tools = common7_tools.parent_path();
```

Comment:
Does a `Path::parent_path(int)` API make sense?

Reply:
I was going to add one but this is the only place parent_path is used repeatedly. I could see an argument for a "now become your parent"?